src/main/ts/ScriptLoader.ts

```diff
@@ -24,20 +25,23 @@ const createState = (): IStateObj => {
 };
 
 interface ScriptLoader {
-  load: (doc: Document, url: string, callback: callbackFn) => void;
+  load: (doc: Document, url: string, callback: callbackFn, loadTimeout?: number, scriptAttributes?: IProps['scriptAttributes']) => void;
   reinitialize: () => void;
 }
 
 const CreateScriptLoader = (): ScriptLoader => {
   let state: IStateObj = createState();
 
-  const injectScriptTag = (scriptId: string, doc: Document, url: string, callback: callbackFn) => {
-    const scriptTag = doc.createElement('script');
+  const injectScriptTag = (scriptId: string, doc: Document, url: string, callback: callbackFn, scriptAttributes?: IProps['scriptAttributes']) => {
+    let scriptTag = doc.createElement('script');
     scriptTag.referrerPolicy = 'origin';
     scriptTag.type = 'application/javascript';
     scriptTag.id = scriptId;
     scriptTag.src = url;
 
+    if (scriptAttributes)
+      scriptTag = { ...scriptTag, ...scriptAttributes }
```

Comment:
We normally wouldn't do this for a DOM node. In this case to keep with the codestyle it will need to be switched to two assignments if you want to support both async and defer. I'm guessing that defer is unlikely to do anything useful being added dynamically like this but I guess it doesn't hurt.